src/librustc_typeck/check/op.rs

```diff
@@ -334,9 +334,17 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
 
                             if !lhs_expr.span.eq(&rhs_expr.span) {
                                 err.span_label(lhs_expr.span, lhs_ty.to_string());
+                                if let FnDef(..) = lhs_ty.sty {
```

Comment:
This logic is identical to the one for `rhs_ty`; refactor into a method and call twice?

Reply:
Thank you the review. I fixed it.